src/sagas/ui.js

```diff
@@ -12,26 +17,39 @@ export function* userDoneTyping() {
   yield put(userDoneTypingAction());
 }
 
-export function* exportGist() {
+function* _githubExport(
+  successAction,
+  failureAction,
+  notDisplayedAction,
+  displayedAction) {
   const exportWindow = yield call(
     openWindowWithWorkaroundForChromeClosingBug,
     `data:text/html;charset=utf-8;base64,${spinnerPage}`,
   );
   const {type, payload: url} =
-    yield take(['GIST_EXPORTED', 'GIST_EXPORT_ERROR']);
+    yield take([successAction, failureAction]);
 
-  if (type === 'GIST_EXPORTED') {
+  if (type === successAction) {
     if (exportWindow.closed) {
-      yield put(gistExportNotDisplayed(url));
+      yield put(notDisplayedAction(url));
     } else {
       exportWindow.location.href = url;
-      yield put(gistExportDisplayed());
+      yield put(displayedAction());
     }
   } else {
     yield call([exportWindow, 'close']);
   }
 }
 
+export function exportGist() {
```

Comment:
I think what you want is a `yield* githubExport`—the `yield*` basically says “here is another generator, put that in charge of generating until it completes”.

I think having a non-generator function that `return`s here might only work as an artifact of the transpilation process—and even if it’s not the case, I’d argue that it’s generally pretty confusing to see a non-generator function `return` a generator.